File: day-01/1.ex
File.read!("./input.txt")
|> String.split("\n\n")
|> Enum.map(fn elf ->
  elf
  |> String.split("\n")
  |> Enum.map(fn elf -> elf |> String.to_integer() end)
  # |> Enum.reduce(fn a, b -> a + b end)
  |> Enum.sum()
end)
|> Enum.max()
|> IO.inspect()

File: day-01/1.ts
export {};

const input = await Deno.readTextFile("./input.txt");

const elves = input.split("\n\n").map((elf) => {
  return elf.split("\n").map((elf) => parseInt(elf));
});

let highest = 0;

for (let i = 0; i < elves.length; i++) {
  const total = elves[i].reduce((a, b) => a + b, 0);

  if (total > highest) {
    highest = total;
  }
}

console.log(highest);
